packages/encoder/lib/adapter.ts

```diff
@@ -172,7 +175,9 @@ export class ProviderAdapter {
             if (error) {
               return reject(error);
             }
-
+            if (response.error) {
```

Comment:
What is this for? Is this to handle the various errors that different clients might throw?  I think you've missed some in that case.